src/components/scenes/ManageTokensScene.js

```diff
@@ -0,0 +1,183 @@
+  const sortedTokenIds = useMemo(() => {
+    // Make a table of preferred tokenId's:
+    const preferredIds = new Set<string>()
+    for (const currencyCode of PREFERRED_TOKENS) {
+      const tokenId = Object.keys(allTokens).find(tokenId => allTokens[tokenId].currencyCode === currencyCode)
+      if (tokenId != null) preferredIds.add(tokenId)
+    }
+
+    return Object.keys(allTokens).sort((id1, id2) => {
+      const token1 = allTokens[id1]
+      const token2 = allTokens[id2]
+      if (preferredIds.has(id1) && !preferredIds.has(id2)) return -1
+      if (!preferredIds.has(id1) && preferredIds.has(id2)) return 1
+      if (token1.currencyCode < token2.currencyCode) return -1
+      if (token1.currencyCode > token2.currencyCode) return 1
+      return 0
+    })
+  }, [allTokens])
+
+  // Filter the list of tokens based on the search term:
+  const filteredTokenIds = useMemo(() => {
+    const target = normalizeForSearch(searchValue)
+    return sortedTokenIds.filter(tokenId => {
+      const token = allTokens[tokenId]
+      const currencyCode = normalizeForSearch(token.currencyCode)
+      const displayName = normalizeForSearch(token.displayName)
+      return currencyCode.includes(target) || displayName.includes(target)
+    })
+  }, [allTokens, searchValue, sortedTokenIds])
+
+  // Shows the wallet picker modal:
+  const handleSelectWallet = async () => {
+    const allowedCurrencyCodes = Object.keys(account.currencyConfig)
+      .filter(pluginId => Object.keys(account.currencyConfig[pluginId].builtinTokens ?? {}).length > 0)
+      .map(pluginId => account.currencyConfig[pluginId].currencyInfo.currencyCode)
+
+    const { walletId, currencyCode } = await Airship.show(bridge => (
+      <WalletListModal
+        allowedCurrencyCodes={allowedCurrencyCodes}
+        // excludeWalletIds={getWalletIdsIfNotTokens()}
+        bridge={bridge}
+        headerTitle={s.strings.select_wallet}
+      />
+    ))
+
+    if (walletId != null && currencyCode != null) {
+      navigation.setParams({ walletId })
+    }
+  }
+
+  return (
+    <SceneWrapper>
+      <SceneHeader underline>
+        <TouchableOpacity onPress={handleSelectWallet}>
+          <Title
+            leftIcon={<CurrencyIcon sizeRem={1.5} walletId={wallet.id} />}
+            rightIcon={<FontAwesomeIcon name="angle-right" size={theme.rem(2)} style={styles.rightIcon} />}
+            text={walletName}
+          />
+        </TouchableOpacity>
+        <EdgeText style={styles.subTitle}>{s.strings.managetokens_top_instructions}</EdgeText>
+        <OutlinedTextInput
+          label={s.strings.search_tokens}
+          marginRem={[1, 2, 0, 1]}
+          returnKeyType="search"
+          searchIcon
+          value={searchValue}
+          onChangeText={setSearchValue}
+        />
+      </SceneHeader>
+      <FlatList
+        data={filteredTokenIds}
+        extraData={allTokens}
+        keyExtractor={tokenId => tokenId}
+        renderItem={(item: FlatListItem<string>) => {
```

Comment:
Declare function in the component body with `useCallback` or the FlatList will always re-render